src/test/java/com/github/dockerjava/cmd/CommitCmdIT.java

```diff
@@ -75,13 +81,14 @@ public void commitWithLabels() throws DockerException {
         InspectImageResponse inspectImageResponse = dockerRule.getClient().inspectImageCmd(imageId).exec();
         LOG.info("Image Inspect: {}", inspectImageResponse.toString());
 
-        // docker bug?
-        if (!getVersion(dockerRule.getClient()).isGreaterOrEqual(RemoteApiVersion.VERSION_1_24)) {
-            Map<String, String> responseLabels = inspectImageResponse.getContainerConfig().getLabels();
-            assertThat(responseLabels.size(), is(2));
-            assertThat(responseLabels.get("label1"), equalTo("abc"));
-            assertThat(responseLabels.get("label2"), equalTo("123"));
-        }
+        //use config here since containerConfig contains the configuration of the container which was
+        //committed to the container
+        //https://stackoverflow.com/questions/36216220/what-is-different-of-config-and-containerconfig-of-docker-inspect
+        Map<String, String> responseLabels = inspectImageResponse.getConfig().getLabels();
+        //swarm will attach additional labels here
+        assertThat(responseLabels.size(), greaterThanOrEqualTo(2));
+        assertThat(responseLabels.get("label1"), equalTo("abc"));
+        assertThat(responseLabels.get("label2"), equalTo("123"));
```

Comment:
but why it work before?

Reply:
It works before because on some docker versions,  the labels was copied to the containerConfig. That was the reason why this test fails on some test configurations. We can only expect the labels on the config object since the container was not started with any labels.

Reply:
ok